lms/djangoapps/certificates/models.py

```diff
@@ -1264,3 +1264,43 @@ class Meta:
 
     def __str__(self):
         return str(self.arguments)
+
+@python_2_unicode_compatible
+class CertificateDateOverride(TimeStampedModel):
+    """
+    Model to manually override a given certificate date with the given date.
+
+    .. no_pii:
+    """
+    generated_certificate = models.OneToOneField(
+        GeneratedCertificate,
+        on_delete=models.CASCADE,
+        primary_key=True,
+        related_name='date_override',
+        help_text="The id of the Generated Certificate to override",
+    )
+    date = models.DateField(
+        help_text="The date to display on the certificate",
+    )
+    reason = models.TextField(
+        help_text="The reason why you are overriding the certificate date (Update this when you add OR edit the date.)",
+    )
+    overridden_by = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        help_text="The last person to save this record",
+    )
```

Comment:
Note that `related_name` is not set on this relational field (compare to the `generated_certificate` `OneToOneField`). By default, this means that the `User` object will have access to the associated `CertificateDateOverride` objects. Do we want that?

- "If you do not specify the `related_name` argument ..., Django will use the lowercase name of the current model as default value." (i.e., We’ll be able to get `user.certificate_date_overrides`. This doesn’t strike me as particularly useful, as the `user` in this case is the admin user who could have lots of these changes)
- "If you’d prefer Django not to create a backwards relation, set `related_name` to `'+'` or end it with `'+'`. For example, this will ensure that the `User` model won’t have a backwards relation to this model:

    ```python
    user = models.ForeignKey(
        User,
        on_delete=models.CASCADE,
        related_name='+',
    )
    ```

- Docs: [https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.ForeignKey.related_name](https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.ForeignKey.related_name)

Reply:
From discussion at Standup: Nah, keep the backwards relation. We’ll probably never use it, but no need to remove it.